pages/hold/request/[bibId-itemId-itemSource]/index.tsx

```diff
@@ -0,0 +1,132 @@
+import Head from "next/head"
+
+import { extractFeatures } from "../../../../src/utils/appUtils"
+import { Heading } from "@nypl/design-system-react-components"
+import { fetchBib } from "../../../../src/utils/bibUtils"
+import { LibraryItem } from "../../../api/hold/confirmation/[bibId-itemId]"
+import isAeonUrl from "../../../../src/utils/aeonLinkUtils"
+import User from "../../../../src/utils/userUtils"
+import {
+  getDeliveryLocations,
+  mapLocationDetails,
+  modelDeliveryLocationName,
+} from "../../../../src/utils/holdUtils"
+import { appConfig } from "../../../../src/config/config"
+
+export default function Request() {
+  return (
+    <>
+      <Head>
+        <title>Request</title>
+      </Head>
+      <Heading>Request</Heading>
+    </>
+  )
+}
+
+/**
+ * newHoldRequest()
+ * The function to return the bib and item data with its delivery locations
+ * to the hold request route.
+ *
+ * This is based on `newHoldRequest` in DFE.
+ */
+async function newHoldRequest(req, res) {
+  const paramString = req.query["bibId-itemId-itemSource"] || ""
+  const [bibId, itemId, itemSource] = paramString.split("-")
+  const bibAndItemId = (bibId || "") + (itemId ? `-${itemId}` : "")
+  const patronId = req.patronTokenResponse?.decodedPatron?.sub || ""
+  const { features } = req.query
+  const urlEnabledFeatures = extractFeatures(features)
+  let barcode
+
+  // Retrieve item
+  try {
+    const bibResponseData = await fetchBib(
+      { bibId: bibAndItemId },
+      {},
+      {
+        fetchSubjectHeadingData: false,
+        features: urlEnabledFeatures,
+      }
+    )
+    const { bib } = bibResponseData
+    barcode = LibraryItem.getItem(bib, itemId).barcode
+    const urlIsAeon = bib.items
```

Comment:
@charmingduchess check if Aeon url logic is needed or why we need to redirect.

Reply:
looks like it is addressing requests made in legacy from [this ticket](https://jira.nypl.org/browse/SCC-2963). Can you add a comment in there calling that out?

Reply:
I don't know whether this is true or not. This part is copied over from DFE so I'd have to do more investigation before stating that's true and addresses the ticket.